Gifski/ExportModifiedVideo.swift

```diff
@@ -0,0 +1,232 @@
+	case finished(URL)
+
+	var shouldShowProgress: Bool {
+		switch self {
+		case .idle, .finished:
+			false
+		case .exporting(_, videoIsOverTwentySeconds: let videoIsOverTwentySeconds):
+			videoIsOverTwentySeconds
+		}
+	}
+	var shouldShowFileExporter: Bool {
+		switch self {
+		case .idle, .exporting:
+			false
+		case .finished:
+			true
+		}
+	}
+
+	var isExporting: Bool {
+		switch self {
+		case .exporting:
+			true
+		default:
+			false
+		}
+	}
+
+	var isFinished: Bool {
+		switch self {
+		case .finished:
+			true
+		default:
+			false
+		}
+	}
+}
+
+
+/**
+Convert a source video to an `.mp4` using the same scale, speed, and crop as the exported `.gif`.
+- Returns: Temporary URL of the exported video.
+*/
+func exportModifiedVideo(conversion: GIFGenerator.Conversion) async throws -> URL {
+	let (composition, compositionVideoTrack) = try await createComposition(
+		conversion: conversion
+	)
+	let videoComposition = try await createVideoComposition(
+		compositionVideoTrack: compositionVideoTrack,
+		conversion: conversion
+	)
+	let outputURL = FileManager.default.temporaryDirectory.appendingPathComponent( "\(UUID().uuidString).mp4")
+
+	let presets = AVAssetExportSession.allExportPresets()
+	guard presets.contains(AVAssetExportPresetHighestQuality) else {
+		throw ExportModifiedVideoView.Error.unableToCreateExportSession
+	}
+	guard await AVAssetExportSession.compatibility(ofExportPreset: AVAssetExportPresetHighestQuality, with: composition, outputFileType: .mp4) else {
+		throw ExportModifiedVideoView.Error.unableToCreateExportSession
+	}
+
+	guard let exportSession = AVAssetExportSession(asset: composition, presetName: AVAssetExportPresetHighestQuality) else {
+		throw ExportModifiedVideoView.Error.unableToCreateExportSession
+	}
+	exportSession.videoComposition = videoComposition
+	try await exportSession.export(to: outputURL, as: .mp4)
+	return outputURL
+}
+
+/**
+Creates the mutable composition along with the video track inserted.
+*/
+private func createComposition(
+	conversion: GIFGenerator.Conversion,
+) async throws -> (AVMutableComposition, AVMutableCompositionTrack) {
+	let composition = AVMutableComposition()
+	guard let compositionTrack = composition.addMutableTrack(
+		withMediaType: .video,
+		preferredTrackID: kCMPersistentTrackID_Invalid
+	) else {
+		throw ExportModifiedVideoView.Error.unableToAddCompositionTrack
+	}
+	try compositionTrack.insertTimeRange(
+		try await conversion.exportModifiedVideoTimeRange,
+		of: try await conversion.firstVideoTrack,
+		at: .zero
+	)
+	return (composition, compositionTrack)
+}
+
+/**
+Create an `AVMutableVideoComposition` that will scale, translate, and crop the `compositionVideoTrack`.
+*/
+private func createVideoComposition(
+	compositionVideoTrack: AVMutableCompositionTrack,
+	conversion: GIFGenerator.Conversion
+) async throws -> AVMutableVideoComposition {
+	let videoComposition = AVMutableVideoComposition()
+
```

Comment:
You may have to use `preferredTransform` here to correctly handle videos with metadata that they are rotated.